### PR TITLE
docutils 0.12 (new formula)

### PR DESCRIPTION
**Pull request by catap**

### CODE REVIEW

File: Library/Formula/docutils.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  desc "Text processing system for processing plaintext documentation into useful formats"
+  homepage "http://docutils.sourceforge.net"
+  url "http://downloads.sourceforge.net/project/docutils/docutils/0.12/docutils-0.12.tar.gz"
+  sha256 "c7db717810ab6965f66c8cf0398a98c9d8df982da39b4cd7f162911eb89596fa"
+
+  def install
+    system "python", "setup.py", "install", "--prefix=#{prefix}"
+  end
+end
+
```
Comment (bfontaine) [inline Library/Formula/docutils.rb]: Please remove this empty line.

File: Library/Formula/docutils.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Docutils < Formula
+  desc "Text processing system for processing plaintext documentation into useful formats"
+  homepage "http://docutils.sourceforge.net"
+  url "http://downloads.sourceforge.net/project/docutils/docutils/0.12/docutils-0.12.tar.gz"
+  sha256 "c7db717810ab6965f66c8cf0398a98c9d8df982da39b4cd7f162911eb89596fa"
+
+  def install
+    system "python", "setup.py", "install", "--prefix=#{prefix}"
+  end
```
Comment (bfontaine) [inline Library/Formula/docutils.rb]: Please add a `test do` block.

File: Library/Formula/docutils.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Docutils < Formula
+  desc "Text processing system for processing plaintext documentation into useful formats"
+  homepage "http://docutils.sourceforge.net"
+  url "http://downloads.sourceforge.net/project/docutils/docutils/0.12/docutils-0.12.tar.gz"
```
Comment (bfontaine) [inline Library/Formula/docutils.rb]: Please use `https://` here.

File: Library/Formula/docutils.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Docutils < Formula
+  desc "Text processing system for processing plaintext documentation into useful formats"
```
Comment (bfontaine) [inline Library/Formula/docutils.rb]: Please shorten the description. “`docutils: desc...`” should fit under 80 characters.

**Comment by DomT4** [comment]
Test is failing:

```
==> /usr/local/Cellar/docutils/0.12/bin/rst2man.py /usr/local/Cellar/docutils/0.12/HISTORY.txtx
Unable to open source file for reading:
  InputError: [Errno 2] No such file or directory: '/usr/local/Cellar/docutils/0.12/HISTORY.txtx'
Error: docutils: failed
```

File: Library/Formula/docutils.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Docutils < Formula
+  desc "Text processing system for reStructuredText"
+  homepage "http://docutils.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/docutils/docutils/0.12/docutils-0.12.tar.gz"
+  sha256 "c7db717810ab6965f66c8cf0398a98c9d8df982da39b4cd7f162911eb89596fa"
+
+  def install
+    system "python", "setup.py", "install", "--prefix=#{prefix}"
```
Comment (dunn) [inline Library/Formula/docutils.rb]: We've got a helper for this:
```ruby
system "python", *Language::Python.setup_install_args(prefix)
```

**Comment by catap** [comment]
@DomT4, thanks! It's a typo :)

**Comment by apjanke** [comment]
Hmmm. This seems to work for me when running under `brew test`, but doesn't work under normal usage from the command line.

```
[~/tmp]
$ brew test docutils
Testing docutils
==> Using the sandbox
==> /usr/local/Cellar/docutils/0.12/bin/rst2man.py /usr/local/Cellar/docutils/0.12/HISTORY.txt
[~/tmp]
$ rst2man.py /usr/local/Cellar/docutils/0.12/HISTORY.txt
Traceback (most recent call last):
  File "/usr/local/bin/rst2man.py", line 21, in <module>
    from docutils.core import publish_cmdline, default_description
ImportError: No module named docutils.core
[✘ ~/tmp]
$ rst2man.py --version
Traceback (most recent call last):
  File "/usr/local/bin/rst2man.py", line 21, in <module>
    from docutils.core import publish_cmdline, default_description
ImportError: No module named docutils.core
```

Perhaps they need wrapper scripts to set up the Python environment? Looking at the installed program, it drops its modules into `/usr/local/lib/python2.7/site-packages`, which may not be available by default under the system `python`, which is what the shebang lines in these scripts are pointing at.

```
$ /usr/bin/python /usr/local/Cellar/docutils/0.12/bin/rst2man.py --help
Traceback (most recent call last):
  File "/usr/local/Cellar/docutils/0.12/bin/rst2man.py", line 21, in <module>
    from docutils.core import publish_cmdline, default_description
ImportError: No module named docutils.core
[✘ ~/tmp]
$ /usr/local/bin/python /usr/local/Cellar/docutils/0.12/bin/rst2man.py --help | head
Usage
=====
  rst2man.py [options] [<source> [<destination>]]
```

**Comment by catap** [comment]
@apjanke this is strange, because it works for me:
```sh
➜  /tmp  rst2man.py /usr/local/Cellar/docutils/0.12/HISTORY.txt | tail -n 4
.SH COPYRIGHT
This document has been placed in the public domain.
.\" Generated by docutils manpage writer.
.
➜  /tmp  
```
What a python version do you use and is it from system or brew?

**Comment by apjanke** [comment]
`/usr/bin/python` is the system Python, version 2.7.5. This is what's in the shebang in the docutils binaries that got installed.
`/usr/local/bin/python` is the Homebrewed Python, version 2.7.10.

```
$ head -1 /usr/local/Cellar/docutils/0.12/bin/*.py
==> /usr/local/Cellar/docutils/0.12/bin/rst2html.py <==
#!/usr/bin/python

==> /usr/local/Cellar/docutils/0.12/bin/rst2latex.py <==
#!/usr/bin/python

==> /usr/local/Cellar/docutils/0.12/bin/rst2man.py <==
#!/usr/bin/python
```

This is on OS X 10.9.5.

I don't know why it's working under `brew test`. Someone who knows more about `brew` & Python – are we doing something to get `$(brew --prefix)/lib/pythonX.Y/site-packages` on the Python search path, or get it to to use the brewed Python even when the shebang line is `#!/usr/bin/python`?

**Comment by mikemcquaid** [comment]
> I don't know why it's working under brew test. Someone who knows more about brew & Python – are we doing something to get $(brew --prefix)/lib/pythonX.Y/site-packages on the Python search path, or get it to to use the brewed Python even when the shebang line is #!/usr/bin/python?

CC @tdsmith as the resident Python expert.

**Comment by tdsmith** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
Migrated to Homebrew/homebrew-core#910

**Comment by apjanke** [comment]
Merged in the migrated PR. Thank you for your contribution to Homebrew!

**Pull request closed without merging**